lib/deps/md5.js

```diff
@@ -7,7 +7,7 @@ var MD5_CHUNK_SIZE = 32768;
 
 function sliceShim(arrayBuffer, begin, end) {
   if (typeof arrayBuffer.slice === 'function') {
-    if (!begin) {
+    if (!begin && !end) {
```

Comment:
^ this was the `sliceShim` bug. if you pass in 0 for `begin` and non-zero for `end`, then it will just call `slice()`, which is wrong.